### PR TITLE
Completed 108

### PR DESCRIPTION
**Pull request by makfazlic**

### CODE REVIEW

File: app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt [hunk 1]
```diff
@@ -263,8 +294,41 @@ class ProfileActivity : BaseActivity(), CoroutineScope {
                     textViews.add(textView)
                     top += 10
                 }
+
+                // Set the number of habits and the average repetitions per week
+                averageRepetitionsPerWeek = daysCount.sum() / daysCount.size
+                habitCountText.text = "Posted habits: $numberOfhabits"
```
Comment (davidbojkovski) [inline app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt:300]: hardcoded string, you can use it with strings.xml with parameters. You can check how it is used in the case of "create_habit_end_time_text" strings resource

File: app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt [hunk 2]
```diff
@@ -263,8 +294,41 @@ class ProfileActivity : BaseActivity(), CoroutineScope {
...
+                avgPerWeekText.text = "Avg. Days in Week: $averageRepetitionsPerWeek"
```
Comment (davidbojkovski) [inline app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt:301]: hardcoded string, you can use it with strings.xml with parameters. You can check how it is used in the case of "create_habit_end_time_text" strings resource

Comment (makfazlic) [reply app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt]: Done, pushed with the changes.

**Review by davidbojkovski — APPROVED** [review]

File: app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt [hunk 3]
```diff
@@ -331,6 +395,31 @@ class ProfileActivity : BaseActivity(), CoroutineScope {
             }
         })
 
+        val refFollow = db.child("users/${uid}/followingPath")
+        refFollow.addListenerForSingleValueEvent(object : ValueEventListener {
+            override fun onDataChange(dataSnapshot: DataSnapshot) {
+                numberOfFollowing = dataSnapshot.childrenCount.toInt()
+                followingText.text = "Following: $numberOfFollowing"
```
[the remark this answers is not in the document]
Comment (makfazlic) [reply app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt]: Done now

File: app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt [hunk 4]
```diff
@@ -263,8 +294,41 @@ class ProfileActivity : BaseActivity(), CoroutineScope {
...
+                    val minute = ends[i].split(":")[1].toInt()
+                    if (hour > latestHour || (hour == latestHour && minute > latestMinute)) {
+                        latestHour = hour
+                        latestMinute = minute
+                    }
+                }
+
+                // Set the earliest and latest habit
+                earliestText.text = "Earliest start: $earliestHour:$earliestMinute"
+                latestText.text = "Latest end: $latestHour:$latestMinute"
```
[the remark this answers is not in the document]
Comment (makfazlic) [reply app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt]: Done now

File: app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt [hunk 5]
```diff
@@ -263,8 +294,41 @@ class ProfileActivity : BaseActivity(), CoroutineScope {
...
+                    val hour = ends[i].split(":")[0].toInt()
+                    val minute = ends[i].split(":")[1].toInt()
+                    if (hour > latestHour || (hour == latestHour && minute > latestMinute)) {
+                        latestHour = hour
+                        latestMinute = minute
+                    }
+                }
+
+                // Set the earliest and latest habit
+                earliestText.text = "Earliest start: $earliestHour:$earliestMinute"
```
[the remark this answers is not in the document]
Comment (makfazlic) [reply app/src/main/java/com/github/mateo762/myapplication/profile/ProfileActivity.kt]: Done now

**Pull request merged**